nupic/encoders/random_distributed_scalar.py

```diff
@@ -131,15 +141,14 @@ def __init__(self, resolution, w=21, n=400, name=None, offset=None,
     # The largest overlap we allow for non-adjacent encodings
     self._maxOverlap = 2
 
-    self.random = numpy.random.RandomState()
-    if seed != -1:
-      self.random.seed(seed)
+    # initialize the random number generators
+    self._seed(seed)
 
     # Internal parameters for bucket mapping
     self.minIndex = None
     self.maxIndex = None
     self._offset = None
-    self._initializeBucketMap(1000, offset)
+    self._initializeBucketMap(MAX_BUCKETS, offset)
```

Comment:
Is `1000` really the max buckets? It looks like the number of initially-generated buckets but can grow indefinitely with subsequent data. Is there a reason to include max buckets in the capnp schema?